### ISSUE TITLE
随机种子设置不起作用

### ISSUE THREAD

**Issue by cloud2009** [PaddlePaddle/Paddle#57745]
### bug描述 Describe the Bug

固定随机种子：

random.seed(2021)
paddle.seed(2021)
np.random.seed(2021)

**但是无法取得一致的网络初始化结果**

网络设置：

seq_len = 336
label_len = 0
pred_len = 96
batch_size = 32
num_workers = 10
enc_in = 7
s = 24

import paddle
import paddle.nn as nn
import numpy as np

class MLP_encoder(nn.Layer):
    def __init__(self, seq_len, s, hidden_len=2048):
        super(MLP_encoder, self).__init__()
        self.seq_len = seq_len
        self.hidden_len = hidden_len
        self.s = s
        self.split = nn.Conv1D(in_channels=1, out_channels=self.s, kernel_size=self.s, stride=self.s)
        self.cat = nn.Conv1DTranspose(in_channels=self.s, out_channels=1, kernel_size=self.s, stride=self.s)

    def forward(self, x): # [Batch, Channel, Input length]
        B, F, L = x.shape
        x = paddle.reshape(x, [B * F, 1, L])
        x = self.split(x)
        x = self.cat(x)
        x = paddle.reshape(x, [B, F, L])
        return x # [Batch, Channel, Input length]
